compiler/rustc_parse/src/parser/item.rs

```diff
@@ -93,13 +93,13 @@ impl<'a> Parser<'a> {
 pub(super) type ItemInfo = (Ident, ItemKind);
 
 impl<'a> Parser<'a> {
-    pub fn parse_item(&mut self) -> PResult<'a, Option<P<Item>>> {
-        self.parse_item_(|_| true).map(|i| i.map(P))
+    pub fn parse_item(&mut self, force_collect: bool) -> PResult<'a, Option<P<Item>>> {
```

Comment:
Can we make `force_collect` a new `enum`? Boolean flags are an antipattern (I have been guilty of using them in the past too.)